File: Algorithms/0001_Two_Sum.py
#! /usr/bin/env python
# coding: utf-8


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        partners = {}
        for i, n in enumerate(nums):
            partner = target - n
            if partner in partners:
                return [partners[partner], i]
            else:
                partners[n] = i

File: Algorithms/0001_Two_Sum.java
import java.util.HashMap; 

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] rs = {0, 0};
        for (int i = 0; i < nums.length; i++){
            if (map.containsKey(target - nums[i])){
                rs[0] = map.get(target - nums[i]);
                rs[1] = i;
                break;
            } else {
                map.put(nums[i], i);
            }
        }
        return rs;
    }
}
